E2e test TestPodConnectivityAfterAntreaRestart is flaky

**Describe the bug**
`TestPodConnectivityAfterAntreaRestart` seems to be flaky, at least when running tests on Kind. I will monitor and check if this also happens in Jenkins tests. On Kind, this seems to happen regardless of the encapsulation mode, and the test logs typically look like this:
```
2020-03-24T19:15:23.3339925Z --- FAIL: TestPodConnectivityAfterAntreaRestart (55.18s)
2020-03-24T19:15:23.3340444Z ##[error]    fixtures.go:55: Creating K8s clientset
2020-03-24T19:15:23.3341200Z ##[error]    fixtures.go:61: Creating 'antrea-test' K8s Namespace
2020-03-24T19:15:23.3341617Z ##[error]    fixtures.go:38: Applying Antrea YAML
2020-03-24T19:15:23.3342297Z ##[error]    fixtures.go:42: Waiting for all Antrea DaemonSet Pods
2020-03-24T19:15:23.3342718Z ##[error]    fixtures.go:46: Checking CoreDNS deployment
2020-03-24T19:15:23.3343525Z ##[error]    connectivity_test.go:104: Creating busybox test Pods 'test-pod-0-4x0w77fw' on 'kind-control-plane'
2020-03-24T19:15:23.3344551Z ##[error]    connectivity_test.go:104: Creating busybox test Pods 'test-pod-1-jj1580ia' on 'kind-worker'
2020-03-24T19:15:23.3345363Z ##[error]    connectivity_test.go:29: Waiting for Pods to be ready and retrieving IPs
2020-03-24T19:15:23.3346232Z ##[error]    connectivity_test.go:38: Retrieved all Pod IPs: map[test-pod-0-4x0w77fw:10.10.0.10 test-pod-1-jj1580ia:10.10.1.5]
2020-03-24T19:15:23.3346792Z ##[error]    connectivity_test.go:47: Ping mesh test between all Pods
2020-03-24T19:15:23.3347595Z ##[error]    connectivity_test.go:56: Ping 'test-pod-0-4x0w77fw' -> 'test-pod-1-jj1580ia': OK
2020-03-24T19:15:23.3348225Z ##[error]    connectivity_test.go:56: Ping 'test-pod-1-jj1580ia' -> 'test-pod-0-4x0w77fw': OK
2020-03-24T19:15:23.3348975Z ##[error]    connectivity_test.go:148: Deleting Antrea Agent DaemonSet
2020-03-24T19:15:23.3349338Z ##[error]    connectivity_test.go:153: Applying Antrea YAML
2020-03-24T19:15:23.3350271Z ##[error]    connectivity_test.go:163: Waiting for all Antrea DaemonSet Pods
2020-03-24T19:15:23.3350644Z ##[error]    connectivity_test.go:169: Restarting CoreDNS Pods
2020-03-24T19:15:23.3351134Z ##[error]    connectivity_test.go:29: Waiting for Pods to be ready and retrieving IPs
2020-03-24T19:15:23.3351847Z ##[error]    connectivity_test.go:38: Retrieved all Pod IPs: map[test-pod-0-4x0w77fw:10.10.0.10 test-pod-1-jj1580ia:10.10.1.5]
2020-03-24T19:15:23.3352247Z ##[error]    connectivity_test.go:47: Ping mesh test between all Pods
2020-03-24T19:15:23.3352903Z ##[error]    connectivity_test.go:54: Ping 'test-pod-0-4x0w77fw' -> 'test-pod-1-jj1580ia': ERROR (command terminated with exit code 1)
2020-03-24T19:15:23.3353878Z ##[error]    connectivity_test.go:56: Ping 'test-pod-1-jj1580ia' -> 'test-pod-0-4x0w77fw': OK
2020-03-24T19:15:23.3354465Z ##[error]    fixtures.go:203: Deleting Pod 'test-pod-0-4x0w77fw'
2020-03-24T19:15:23.3355031Z ##[error]    fixtures.go:203: Deleting Pod 'test-pod-1-jj1580ia'
2020-03-24T19:15:23.3355688Z ##[error]    fixtures.go:89: Exporting test logs to '/tmp/antrea-test-359293269/TestPodConnectivityAfterAntreaRestart.Mar24-19-15-13'
2020-03-24T19:15:23.3356250Z ##[error]    fixtures.go:196: Deleting 'antrea-test' K8s Namespace
```
It is surprising that the ping works in one direction and not the other. Typically the ping we run first is the one that consistently fails, so maybe it is a timing issue.

--- a/test/e2e/framework.go
+++ b/test/e2e/framework.go
@@ -268,10 +268,13 @@ func (data *TestData) deleteTestNamespace(timeout time.Duration) error {
 func (data *TestData) deployAntreaCommon(yamlFile string, extraOptions string) error {
 	// TODO: use the K8s apiserver when server side apply is available?
 	// See https://kubernetes.io/docs/reference/using-api/api-concepts/#server-side-apply
-	rc, stdout, _, err := provider.RunCommandOnNode(masterNodeName(), fmt.Sprintf("kubectl apply %s -f %s", extraOptions, yamlFile))
+	rc, _, _, err := provider.RunCommandOnNode(masterNodeName(), fmt.Sprintf("kubectl apply %s -f %s", extraOptions, yamlFile))
 	if err != nil || rc != 0 {
-		fmt.Println(stdout)
 		return fmt.Errorf("error when deploying Antrea; is %s available on the master Node?", yamlFile)
+	}
+	rc, _, _, err = provider.RunCommandOnNode(masterNodeName(), fmt.Sprintf("kubectl -n %s rollout status ds/%s --timeout=%v", antreaNamespace, antreaDaemonSet, defaultTimeout))
+	if err != nil || rc != 0 {
+		return fmt.Errorf("error when waiting for Antrea rollout to complete")
 	}
 	return nil
 }
@@ -295,10 +298,14 @@ func (data *TestData) waitForAntreaDaemonSetPods(timeout time.Duration) error {
 			return false, fmt.Errorf("error when getting Antrea daemonset: %v", err)
 		}
 
-		// Make sure that all Daemon Pods are available and that there is no ongoing rolling
-		// update.
-		if daemonSet.Status.NumberAvailable == daemonSet.Status.DesiredNumberScheduled &&
-			daemonSet.Status.UpdatedNumberScheduled == daemonSet.Status.DesiredNumberScheduled {
+		// Make sure that all Daemon Pods are available.
+		// We use clusterInfo.numNodes instead of DesiredNumberScheduled because
+		// DesiredNumberScheduled may not be updated right away. If it is still set to 0 the
+		// first time we get the DaemonSet's Status, we would return immediately instead of
+		// waiting.
+		desiredNumber := int32(clusterInfo.numNodes)
+		if daemonSet.Status.NumberAvailable == desiredNumber &&
+			daemonSet.Status.UpdatedNumberScheduled == desiredNumber {
 			// Success
 			return true, nil
 		}

